File: src/main/java/editor/gui/ccp/data/DataFlavors.java
package editor.gui.ccp.data;

import java.awt.datatransfer.DataFlavor;

import editor.collection.deck.CategorySpec;
import editor.database.card.Card;

/**
 * Collection of data flavors used for drag-and-drop in the editor.
 */
public interface DataFlavors
{
	/**
	 * Data flavor representing cards being transferred.
	 */
	DataFlavor cardFlavor = new DataFlavor(DataFlavor.javaJVMLocalObjectMimeType + ";class=\"" + Card[].class.getName() + "\"", "Card Array");

	DataFlavor categoryFlavor = new DataFlavor(DataFlavor.javaJVMLocalObjectMimeType + ";class=\"" + CategorySpec.class.getName() + "\"", "Category Specification");

	/**
	 * Data flavor representing entries in a deck.  Transfer data will appear as a
	 * map of cards onto an integer representing the number of copies to transfer.
     * It will also have auxiliary data including the source EditorFrame and ID.
	 */
    DataFlavor entryFlavor = new DataFlavor(DataFlavor.javaJVMLocalObjectMimeType + ";class=\"" + EntryTransferData.class.getName() + "\"", "Deck Entries");
}

File: src/main/scala/editor/gui/ccp/data/DataFlavors.scala
package editor.gui.ccp.data

import editor.collection.deck.Category
import editor.database.card.Card

import java.awt.datatransfer.DataFlavor

/**
 * Collection of data flavors used for copying data.
 * @author Alec Roelke
 */
object DataFlavors {
  /** Data flavor used for copying cards, typically from the inventory panel. */
  lazy val cardFlavor = DataFlavor(s"${DataFlavor.javaJVMLocalObjectMimeType};class=\"${classOf[Array[Card]].getName}\"", "Card Array")

  /** Data flavor used for copying cards and metadata, typically from a deck table. */
  lazy val entryFlavor = DataFlavor(s"${DataFlavor.javaJVMLocalObjectMimeType};class=\"${classOf[EntryTransferData].getName}\"", "Deck Entries")

  /** Data flavor used for copying categorizations and (TBD) cards in catgories. */
  lazy val categoryFlavor = DataFlavor(s"${DataFlavor.javaJVMLocalObjectMimeType};class=\"${classOf[Category].getName}\"", "Categorization")
}